src/tools/agent-ctl/src/vm/mod.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2024 Microsoft Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Description: Boot UVM for testing container storages/volumes.
+
+use anyhow::{anyhow, Context, Result};
+use hypervisor::Hypervisor;
+use kata_types::config::{hypervisor::HYPERVISOR_NAME_CH, hypervisor::HYPERVISOR_NAME_QEMU};
+use slog::info;
+use std::sync::Arc;
+
+mod vm_ops;
+mod vm_utils;
+
+#[derive(Clone)]
+pub struct TestVm {
+    pub hypervisor_name: String,
+    pub hypervisor_instance: Arc<dyn Hypervisor>,
+    pub socket_addr: String,
+    pub hybrid_vsock: bool,
+}
+
+// Helper method to boot a test pod VM
+pub fn setup_vm(hypervisor_name: &str) -> Result<TestVm> {
+    info!(
+        sl!(),
+        "booting a pod vm using hypervisor:{:?}", hypervisor_name
+    );
+
+    // only supports qemu & cloud hypervisor for now
+    if !hypervisor_name.contains(HYPERVISOR_NAME_QEMU)
+        && !hypervisor_name.contains(HYPERVISOR_NAME_CH)
```

Comment:
what's about using a support-list for `vec![HYPERVISOR_NAME_QEMU, HYPERVISOR_NAME_CH]` to extend it easily in future if firecracker is in the TODO list.